include/fem-dem/void_fraction.h

```diff
@@ -133,9 +133,18 @@ class VoidFractionBase : public PhysicsLinearSubequationsSolver
     else
       {
         // Usual case, for quad/hex meshes
-        fe         = std::make_shared<FE_Q<dim>>(fe_degree);
-        mapping    = std::make_shared<MappingQ<dim>>(fe->degree);
-        quadrature = std::make_shared<QGauss<dim>>(fe->degree + 1);
+        fe      = std::make_shared<FE_Q<dim>>(fe_degree);
+        mapping = std::make_shared<MappingQ<dim>>(fe->degree);
+        if (this->void_fraction_parameters->qcm_quadrature_rule ==
+            Parameters::VoidFractionQCMRule::gauss)
+          quadrature = std::make_shared<QGauss<dim>>(
+            this->void_fraction_parameters->qcm_n_quadrature_points);
+        else if (this->void_fraction_parameters->qcm_quadrature_rule ==
+                 Parameters::VoidFractionQCMRule::gauss_lobatto)
+          quadrature = std::make_shared<QGaussLobatto<dim>>(
+            this->void_fraction_parameters->qcm_n_quadrature_points);
```

Comment:
You need to assert that the number of quadrature point is positive
Or what you could do is if it's -1 you put degree+1 and if it's a positive number you use that number.
Here nothing prevents you from putting a negative value.